recipes/libwebp/all/conanfile.py

```diff
@@ -4,87 +4,95 @@
+        self._cmake.definitions["WEBP_ENABLE_SIMD"] = self.options.with_simd
         if self._version_components[0] >= 1:
-            cmake.definitions["WEBP_NEAR_LOSSLESS"] = self.options.near_lossless
+            self._cmake.definitions["WEBP_NEAR_LOSSLESS"] = self.options.near_lossless
         else:
-            cmake.definitions["WEBP_ENABLE_NEAR_LOSSLESS"] = self.options.near_lossless
-        cmake.definitions['WEBP_ENABLE_SWAP_16BIT_CSP'] = self.options.swap_16bit_csp
+            self._cmake.definitions["WEBP_ENABLE_NEAR_LOSSLESS"] = self.options.near_lossless
+        self._cmake.definitions["WEBP_ENABLE_SWAP_16BIT_CSP"] = self.options.swap_16bit_csp
         # avoid finding system libs
-        cmake.definitions['CMAKE_DISABLE_FIND_PACKAGE_GIF'] = True
-        cmake.definitions['CMAKE_DISABLE_FIND_PACKAGE_PNG'] = True
-        cmake.definitions['CMAKE_DISABLE_FIND_PACKAGE_TIFF'] = True
-        cmake.definitions['CMAKE_DISABLE_FIND_PACKAGE_JPEG'] = True
-        cmake.definitions['WEBP_BUILD_ANIM_UTILS'] = False
-        cmake.definitions['WEBP_BUILD_CWEBP'] = False
-        cmake.definitions['WEBP_BUILD_DWEBP'] = False
-        cmake.definitions['WEBP_BUILD_IMG2WEBP'] = False
-        cmake.definitions['WEBP_BUILD_GIF2WEBP'] = False
-        cmake.definitions['WEBP_BUILD_VWEBP'] = False
-        cmake.definitions['WEBP_BUILD_EXTRAS'] = False
-        cmake.definitions['WEBP_BUILD_WEBPINFO'] = False
-        cmake.definitions['WEBP_BUILD_WEBPMUX'] = False
+        self._cmake.definitions["CMAKE_DISABLE_FIND_PACKAGE_GIF"] = True
+        self._cmake.definitions["CMAKE_DISABLE_FIND_PACKAGE_PNG"] = True
+        self._cmake.definitions["CMAKE_DISABLE_FIND_PACKAGE_TIFF"] = True
+        self._cmake.definitions["CMAKE_DISABLE_FIND_PACKAGE_JPEG"] = True
+        self._cmake.definitions["WEBP_BUILD_ANIM_UTILS"] = False
+        self._cmake.definitions["WEBP_BUILD_CWEBP"] = False
+        self._cmake.definitions["WEBP_BUILD_DWEBP"] = False
+        self._cmake.definitions["WEBP_BUILD_IMG2WEBP"] = False
+        self._cmake.definitions["WEBP_BUILD_GIF2WEBP"] = False
+        self._cmake.definitions["WEBP_BUILD_VWEBP"] = False
+        self._cmake.definitions["WEBP_BUILD_EXTRAS"] = False
+        self._cmake.definitions["WEBP_BUILD_WEBPINFO"] = False
+        self._cmake.definitions["WEBP_BUILD_WEBPMUX"] = False
 
-        cmake.configure()
+        self._cmake.configure()
 
-        return cmake
+        return self._cmake
 
     def build(self):
-        tools.patch(base_path=self._source_subfolder, patch_file='0001-fix-dll-export.patch')
-        tools.patch(base_path=self._source_subfolder, patch_file='0002-enable-webpmux.patch')
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+        # build libwebpmux (not in patch file to avoid one patch per version)
+        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
```

Comment:
Why did you remove that option and is forcing it now? As optional from author project, you should give the same proposal, do not force an option.

Reply:
Which option?
The removed patch added an option in CMakeLists unused in conanfile and any file in libwebp (before or after patch), and removed an option set anyway to false by conanfile. The others modifications of this patch had the same effect than the simple replace_in_file proposed here.

Reply:
> Which option?

WEBP_BUILD_WWEBP

> The removed patch added an option in CMakeLists unused in conanfile and any file in libwebp (before or after patch), and removed an option set anyway to false by conanfile.

I see, so was WWEBP optional by the author or not?

Reply:
It was useless, a typo I think or something which was used before 1.0.3. Take a look to 1.0.3 source code of libwebp and to the old conan recipe.